pkg/volume/gcepd/attacher.go

```diff
@@ -138,6 +141,52 @@ func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Sp
 	return volumesAttachedCheck, nil
 }
 
+
+// search Windows disk number by LUN
+func getDiskID(pdName string, exec mount.Exec) (string, error) {
+	cmd := `Get-GcePdName | select Name, DeviceId | ConvertTo-Json`
+	output, err := exec.Run("powershell", "/c", cmd)
+	if err != nil {
+		klog.Errorf("Get-GcePdName failed, error: %v, output: %q", err, string(output))
+		err = errors.New(err.Error() + " " + string(output))
+		return "", err
+	}
+
+	var data []map[string]interface{}
+	if err = json.Unmarshal(output, &data); err != nil {
+		klog.Errorf("Get-Disk output is not a json array, output: %q", string(output))
+		return "", err
+	}
+
+	for _, pd := range data {
+		if jsonName, ok := pd["Name"]; ok {
+			if name, ok := jsonName.(string); ok {
+				klog.Infof("data name %q", name)
+				if name == pdName {
+					klog.Infof("found the disk %q", name)
+					if diskNum, ok := pd["DeviceId"]; ok {
+						switch v := diskNum.(type) {
```

Comment:
Is this switch really necessary? The type should either be always string or always int here, right?

This seems like future-proofing for a type change that's unlikely to happen.

Reply:
I see in azure, this value is a float64, so I think it might be safer to do this way?